docs/dev.rst

```diff
@@ -0,0 +1,490 @@
+(In this scenario ``npe03``, ``npe04``, and ``npe05`` do not have to be uninstalled to uninstall ``npe01``.)
+
+
+
+Unmanaged Metadata Dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Specify unmanaged metadata to be deployed by specifying a ``zip_url`` and, optionally,
+``subfolder``, ``namespace_inject``, ``namespace_strip``, and ``unmanaged`` under the
+``project__dependencies`` section of the cumulusci.yml file.
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - zip_url: https://SOME_HOST/metadata.zip
+
+When the ``update_dependencies`` task runs, it downloads the zip file and deploys it via the Metadata API.
+The zip file must contain valid metadata for use with a deploy, including a ``package.xml`` file in the root.
+
+
+
+Specify a Subfolder of the Zip File
+***********************************
+
+Use the ``subfolder`` option to specify a subfolder of the zip file to use for the deployment. 
+
+.. tip::
+    
+    This option is handy when referring to metadata stored in a GitHub repository.
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - zip_url: https://github.com/SalesforceFoundation/CumulusReports/archive/master.zip
+              subfolder: CumulusReports-master/record_types
+
+When ``update_dependencies`` runs, it still downloads the zip from ``zip_url``, but then builds a new zip containing only the content of ``subfolder``, starting inside ``subfolder`` as the zip's root.
+
+
+
+Inject Namespace Prefixes
+*************************
+
+CumulusCI has support for tokenizing references to a package's namespace prefix in code.
+When tokenized, all occurrences of the namespace prefix, are replaced with ``%%%NAMESPACE%%%`` inside of files and ``___NAMESPACE___`` in file names.
+
+For more on this topic see :ref:`namespace injection`. 
+
+
+Automatic Cleaning of ``meta.xml`` Files on Deploy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To let CumulusCI fully manage the project's dependencies, the ``deploy`` task (and other tasks based on ``cumulusci.tasks.salesforce.Deploy``, or subclasses of it) automatically removes the ``<packageVersion>`` element and its children from all ``meta.xml`` files in the deployed metadata. Removing these elements does not affect the files on the filesystem.
+
+This feature supports CumulusCI's automatic dependency resolution by avoiding a need for projects to manually update XML files to reflect current dependency package versions.
+
+.. note:: 
+
+    If the metadata being deployed references namespaced metadata that does not exist in the currently installed package, the deployment throws an error as expected.
+
+.. tip:: 
+
+    The automatic cleaning of ``meta.xml`` files can be disabled by setting the ``clean_meta_xml`` option to ``False``.
+
+One drawback of this approach is that developers need to handle the diffs in the ``meta.xml`` files by either ignoring them, or committing them as part of their work in a feature branch. 
```

Comment:
We say above that cleaning doesn't affect the files on the filesystem - I'm confused by this line.

Reply:
I don't think I can speak to how this currently works.

Reply:
This paragraph is no longer accurate and should be removed.